### ISSUE TITLE
Dialog window

### ISSUE THREAD

**Issue by BetterToAutomateTheWorld** [MinetestForFun/server-minetestforfun-hungry_games#39]
Dialog window instead of a message wish flood the chat ? 

https://forum.minetest.net/viewtopic.php?f=11&t=12829&view=unread#p184593

Maybe tweak this mod to inforamtion window instead of "kick window"

Just to say "use /vote, vote block, or vote in your inventory" when players are in the lobby and waiting for a game to start

**Comment by Lymkwi** [comment]
At the moment I can implement that into the server, but I need to remove some few things.
Here is how it would work :
- The previous Hungry Game has ended
- Players wait in the lobby
- Someone triggers the vote (with a `start_vote` command of some sort)
- This vote goes on for a certain duration (something like 2 minutes)
- If the result is a big "nay", then we shall wait until someone else uses the `vote_start`
- If the result is a "yay", we will start the hungry games.
- The loop loops again

That's the only way I can implement this at the moment, but it has many flaws :
- Someone must **manually** start the vote, or a specifically designed system must do it every two minutes starting from a specific moment and considering the previous results
- Even if everybody votes, they would have to wait until the complete span of time
- We do not know how to handle quits, abstainsions.
- We have to choose a percent of "yay" (a too high percent would probably cause no game to start, not enough would cause player not to be ready?)
- **The vote command would disappear**. Instead players would have to use `/yes`, `/no`, or `/abstain`.

**In conclusion**: The engine is currently handling the votes perfectly. Players are used to it, based on their experience on other servers or on our server. `vote` is a too specific action, it's not a real vote, it's a sort of "I"m ready, let's go" thing. We could try to implement that but I don't think it fits the `vote` feature as good as the current method does.

**Comment by Lymkwi** [comment]
![HUD](https://lut.im/YdzJMbjI/ano2zrhv)

I got a copy, I couldn't use rubenwardy's box texture but I put a message upon the HUD cross.
https://github.com/MinetestForFun/hungry_games/commit/5d77227f9ca0da67be3c2d6b5b4961bd4b1c9e71

**Comment by ObaniGemini** [comment]
"Remember to vote using a voteblock or by typing /vote in the chat"

**Comment by Lymkwi** [comment]
@ObaniGemini: Indeed. That was that message players were constantely harrassed with in the chat. That's why we decided to put it in the HUD, to not flood their chatbox.

**Comment by ObaniGemini** [comment]
@LeMagnesium : What I want to tell is just /vote isn't enough clear ^^
        When players see it they cannot know if it's something to type, a name,... (I mean casual players, most part of HG players )

**Comment by Lymkwi** [comment]
Are you telling me that they are just dumb enough to not understand that they need to type "/vote" in the chatbox when they are told to? If they are, just kick them and wait until they are mature and intelligent enough to understand that...

**Comment by ObaniGemini** [comment]
@LeMagnesium : I think you haven't met enough humans on video games

**Comment by Lymkwi** [comment]
I cannot fullfil the needs of impotent people.
Either way, someone will tell them to type /vote sooner or later.
They learn to type "T" and "/vote" or they get out. Period.

**Comment by BetterToAutomateTheWorld** [comment]
Mg, i think the "/vote" way for voting is the last way we need to inform, the most important way are :

**Vote blocks** > **inventory "vote"** > type "/vote"

For a shorter alert, maybe juste tell to the players the 2 first solutions ?

PS : the way you alert seems to be very interresting :) (so the modal window to the right isn't enough interresting or too complicated for you ?)

**Comment by Lymkwi** [comment]
The window on the right was not visible enough, and it was a mess to tweak the positions used in the vote mod, mainly because it was in pixels and not percents, so I started from only the hudkit and put it over the cross.

I think we should do a little test. My idea is to tweak our blocks, the inventory button and /vote. Whenever one of those is pushed, a specific message is sent in the logs. Then, using some grep | wc -l , we'll see which way is the more used by players.
Until we implement that, let's use that message :  https://github.com/MinetestForFun/hungry_games/commit/3bf8e4f19766a408ac8326ebde0ec13ee30a02d9

**Comment by BetterToAutomateTheWorld** [comment]
Actually, if we push one of our solutions to "vote", the message disappears, no need other tweaks.

**Comment by Lymkwi** [comment]
> Actually, if we push one of our solutions to "vote", the message disappears

Of course, like the message was not sent to players who had already voted. If everything is fine then I close this. If you want to test my idea on statistics about what method players use, open another ticket.
